ArcGISRuntimeSDKQt_CppSamples/Maps/ReadGeopackage/ReadGeopackage.cpp

```diff
@@ -0,0 +1,125 @@
+// [WriteFile Name=ReadGeopackage, Category=Maps]
+// [Legal]
+// Copyright 2018 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "ReadGeopackage.h"
+
+#include "Map.h"
+#include "MapQuickView.h"
+
+#include <QDir>
+
+using namespace Esri::ArcGISRuntime;
+using namespace std;
+
+ReadGeopackage::ReadGeopackage(QQuickItem* parent /* = nullptr */):
+  QQuickItem(parent)
+{
+}
+
+void ReadGeopackage::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<ReadGeopackage>("Esri.Samples", 1, 0, "ReadGeopackageSample");
+}
+
+void ReadGeopackage::componentComplete()
+{
+  QQuickItem::componentComplete();
+
+  // find QML MapView component
+  m_mapView = findChild<MapQuickView*>("mapView");
+
+  // Create a map using the topographic basemap
+  m_map = new Map(Basemap::topographic(this), this);
+
+  // Set map to map view
+  m_mapView->setMap(m_map);
+
+  // Create a member variable to contain the layers
+  QList<Layer*> m_layerList;
+
+  // Create the initial view area
+  Viewpoint initialViewpoint(39.7294, -104.8319, 500000);
+  m_mapView->setViewpoint(initialViewpoint);
+
+  // Initialize the read operation
+  listMaps();
+}
+
+// Read the geopackage and create lists for the names and layer data
+void ReadGeopackage::listMaps()
```

Comment:
`listMaps` is probably not what the function should be called.  Consider renaming to `readGeoPackage`